src/ess/reduce/uncertainty.py

```diff
@@ -36,50 +38,99 @@
 
 @overload
 def broadcast_with_upper_bound_variances(
-    data: sc.Variable, sizes: Dict[str, int]
+    data: sc.Variable, prototype: sc.DataArray | sc.Variable
 ) -> sc.Variable:
     pass
 
 
 @overload
 def broadcast_with_upper_bound_variances(
-    data: sc.DataArray, sizes: Dict[str, int]
+    data: sc.DataArray, prototype: sc.DataArray | sc.Variable
 ) -> sc.DataArray:
     pass
 
 
 def broadcast_with_upper_bound_variances(
-    data: Union[sc.Variable, sc.DataArray], sizes: Dict[str, int]
+    data: Union[sc.Variable, sc.DataArray], prototype: sc.DataArray | sc.Variable
 ) -> Union[sc.Variable, sc.DataArray]:
-    if _no_variance_broadcast(data, sizes):
+    """
+    Compute an upper bound for the variances of the broadcasted data.
+
+    The variances of the broadcasted data are computed by scaling the variances of the
+    input data by the volume of the new subspace. The volume of the new subspace is
+    computed as the product of the sizes of the new dimensions.
+
+    Parameters
+    ----------
+    data:
+        The data to broadcast.
+    prototype:
+        Defines the new sizes (dims and shape). If present, masks are used to exclude
+        masked values from the variance computation.
+
+    Returns
+    -------
+    :
+        The data with the variances scaled by the volume of the new subspace.
+    """
+    if _no_variance_broadcast(data, prototype.sizes):
         return data
-    size = 1
+    sizes = {dim: size for dim, size in prototype.sizes.items() if dim not in data.dims}
+    sizes = prototype.sizes
```

Comment:
Is there a typo here? redefining `sizes`?

Reply:
Oops, first line was left over from refactor.